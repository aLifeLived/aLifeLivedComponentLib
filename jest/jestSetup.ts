/* eslint-disable no-undef */
jest.useFakeTimers({ legacyFakeTimers: true }); // Stops reference error from appearing
