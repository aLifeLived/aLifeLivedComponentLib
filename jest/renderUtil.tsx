import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';
import { library } from '@fortawesome/fontawesome-svg-core';
import { iconList } from '../src/themes/systemIcon/IconList';

library.add(iconList);

function render(ui: any, { ...renderOptions } = {}) {
  function Wrapper({ children }: { children: any }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
