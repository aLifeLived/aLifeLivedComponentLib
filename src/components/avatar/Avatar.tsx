import React from 'react';
import FastImage, { Priority, ResizeMode } from 'react-native-fast-image';
import { Box } from '../../themes/box/Box';
import {
  StyledImageProps,
  SystemImage,
} from '../../themes/systemImage/SystemImage';
import { Icon } from '../icon/Icon';

type AvatarProps = {
  resizeMode?: ResizeMode;
  iconWidth?: number;
  iconHeight?: number;
  iconSize?: number;
  source: {
    uri: string;
    priority: Priority;
  };
};

export const Avatar: React.FC<AvatarProps & StyledImageProps> = ({
  source,
  resizeMode = 'cover',
  variant,
  iconWidth,
  iconHeight,
  iconSize,
  ...rest
}) => {
  return !source.uri ? (
    <Box
      borderWidth={1}
      padding="m"
      width={iconWidth}
      height={iconHeight}
      borderRadius="xXXlarge"
      alignItems="center"
      justifyContent="center"
      {...rest}
    >
      <Icon icon="user" size={iconSize} />
    </Box>
  ) : (
    <SystemImage
      source={{
        uri: source.uri,
        priority: FastImage.priority[source.priority],
      }}
      resizeMode={FastImage.resizeMode[resizeMode]}
      variant={variant}
      {...rest}
    />
  );
};
