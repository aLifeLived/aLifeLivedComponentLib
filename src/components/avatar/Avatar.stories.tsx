import React from 'react';
import { storiesOf } from '@storybook/react-native';

// Component
import { Avatar } from './Avatar';
import { Box } from '../../themes/box/Box';

storiesOf('Avatar', module)
  .addDecorator(Story => (
    <Box margin="s" justifyContent="center" alignItems="center">
      {Story()}
    </Box>
  ))
  .add('Round avatar', () => (
    <>
      <Avatar
        source={{ uri: '', priority: 'normal' }}
        variant="roundAvatar"
        margin="s"
      />
      <Avatar
        source={{
          uri: 'https://picsum.photos/id/237/200/300',
          priority: 'normal',
        }}
        variant="roundAvatar"
        margin="s"
      />
    </>
  ));
