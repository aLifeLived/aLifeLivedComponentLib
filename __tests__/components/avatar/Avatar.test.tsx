import React from 'react';
import { Priority } from 'react-native-fast-image';
import { render } from './../../../jest/renderUtil';

// Component
import { Avatar } from '../../../src/components/avatar/Avatar';

const setup = (
  source: {
    uri: string;
    priority: Priority;
  },
  variant: 'roundAvatar',
  iconHeight = 50,
  iconWidth = 50,
  iconSize = 30
) => {
  // Screen
  const utils = render(
    <Avatar
      source={source}
      resizeMode="center"
      variant={variant}
      iconHeight={iconHeight}
      iconWidth={iconWidth}
      iconSize={iconSize}
    />
  );

  return {
    ...utils,
  };
};

describe('components >> avatar ', () => {
  describe('Round avatar', () => {
    test('Should render round avatar correctly', () => {
      const { toJSON } = setup(
        { uri: 'https://picsum.photos/id/237/200/300', priority: 'normal' },
        'roundAvatar'
      );
      expect(toJSON()).toMatchSnapshot();
    });

    test('Should render icon correctly when no source', () => {
      const { toJSON } = setup(
        { uri: null, priority: 'normal' },
        'roundAvatar'
      );
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
