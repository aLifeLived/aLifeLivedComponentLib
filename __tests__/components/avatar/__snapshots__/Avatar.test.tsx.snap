// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components >> avatar  Round avatar Should render icon correctly when no source 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 96,
        "borderWidth": 1,
        "display": "flex",
        "height": 50,
        "justifyContent": "center",
        "padding": 16,
        "width": 50,
      },
      Array [
        Object {},
      ],
    ]
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={30}
    bbWidth={30}
    fill="#000"
    focusable={false}
    height={30}
    meetOrSlice={0}
    minX={0}
    minY={0}
    secondaryFill={null}
    secondaryOpacity={0.4}
    size={30}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {},
        Object {
          "flex": 0,
          "height": 30,
          "width": 30,
        },
      ]
    }
    vbHeight={512}
    vbWidth={448}
    width={30}
  >
    <RNSVGGroup
      fill={4278190080}
      propList={
        Array [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
        fill={4278190080}
        fillOpacity={1}
        propList={
          Array [
            "fill",
            "fillOpacity",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`components >> avatar  Round avatar Should render round avatar correctly 1`] = `
<View
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Array [
        Object {
          "borderRadius": 96,
          "height": 50,
          "width": 50,
        },
      ],
    ]
  }
>
  <FastImageView
    resizeMode="center"
    source={
      Object {
        "priority": "normal",
        "uri": "https://picsum.photos/id/237/200/300",
      }
    }
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</View>
`;
